### PR TITLE
fix: Data ingestion fails due to OpenSearch "documents" index not found exception

### PR DESCRIPTION
**Pull request by mpawlow**
### Issues

- #845
- #973

### Summary

Fix data ingestion failures caused by a race condition where the backend attempted to interact with Langflow and OpenSearch before those services were ready, and improve logging and vector index configuration.

### Bug Fix: Langflow and OpenSearch Readiness

- Add new wait_for_langflow() utility with exponential backoff and jitter to poll Langflow's /health endpoint before proceeding with API key generation and client initialization (src/utils/langflow_utils.py)
- Add LangflowNotReadyError exception raised when Langflow fails to become ready within the retry limit
- Reorder AppClients.initialize() to create the langflow_http_client before calling wait_for_langflow() and get_langflow_api_key(), ensuring the HTTP client exists before health checks begin (src/config/settings.py)
- Move Langflow API key generation (get_langflow_api_key()) to after the Langflow readiness check instead of the very start of initialization
- Guard the onboarding endpoint with wait_for_langflow(), returning HTTP 503 if Langflow is unavailable (src/api/settings.py)
- Extract init_index_when_ready() to explicitly sequence wait_for_opensearch() before init_index(), and call it from the onboarding endpoint (src/main.py)
- Remove the redundant wait_for_opensearch() call inside init_index() so it can be reused without double-waiting
- Add post-onboarding index initialization in startup_tasks() to recreate the OpenSearch index when the volume has been deleted but onboarding was already completed

### Vector Index Configuration

- Increase ef_construction from 100 to 512 across all index creation paths for improved search recall (src/config/settings.py, src/utils/embedding_fields.py, src/utils/embeddings.py, flows/components/opensearch_multimodal.py, scripts/migrate_embedding_model_field.py)

### Logging Improvements

- Add colored log output with ANSI codes for log level and timestamp, respecting NO_COLOR env var and non-TTY environments (src/utils/logging_config.py)
- Document LOG_LEVEL and NO_COLOR environment variables in .env.example

### Code Quality

- Fix trailing whitespace throughout src/api/settings.py (whitespace-only cleanup)

### Testing

- Add comprehensive unit tests for wait_for_langflow() covering success, transient failures, retry exhaustion, backoff bounds, and edge cases (tests/unit/test_langflow_utils.py)
- Add tests/unit/conftest.py with a no-op onboard_system fixture override so unit tests don't require running infrastructure

### Screenshots

2026-02-13 20_40_19-mpawlow@OpenRAG1_ ~_git_openrag

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -123,6 +123,15 @@ NEXT_ALLOWED_DEV_ORIGINS=
 # LANGFLOW_WORKERS=2    # Langflow workers (default: 1)
 # DOCLING_WORKERS=2     # Docling workers (default: 1)
 
+# OPTIONAL: Set the logging level for the Backend service
+# - Supported values: DEBUG, INFO, WARNING, ERROR, CRITICAL
+# - Default: INFO
+# LOG_LEVEL=DEBUG
```
Comment (edwinjosechittilappilly) [inline .env.example:129]: should we add this to docker compose and helm?

Comment (mpawlow) [reply .env.example:129]: Yes, good catch

- Added `LOG_LEVEL` to the relevant Helm / Docker configuration files
- Also, added a whole bunch of missing environment variables that were present in `.env.example` but not in the relevant Helm / Docker configuration files
- For full details, see https://github.com/langflow-ai/openrag/pull/974/changes/b877c384cce3789550c0b1bab1db6a35b2a5e603
